zipline/gens/tradesimulation.py

```diff
@@ -206,46 +207,88 @@ def _process_snapshot(self, dt, snapshot, instant_fill):
         blotter_process_trade = self.algo.blotter.process_trade
         blotter_process_benchmark = self.algo.blotter.process_benchmark
 
-        for event in snapshot:
+        # Containers for the snapshotted events, so that the events are
+        # processed in a predictable order, without relying on the sorted order
+        # of the individual sources.
+
+        # There is only one benchmark per snapshot.
+        benchmark = None
+        # trades are initialized as a list since there is often a trade per
+        # snapshot, so better to allocate the list than check for is None
+        # on each event.
+        trades = []
+        # splits list will be allocated when a split occurs in the snapshot.
+        splits = None
```

Comment:
The extra complexity and boolean checks necessitated by lazily-allocating these seem not worth it to me.